src/CoreWlan/CWKeychain.cs

```diff
@@ -0,0 +1,231 @@
+using System;
+using System.Runtime.InteropServices;
+
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+using Security;
+
+#if NET
+using System.Runtime.Versioning;
+#endif
+
+
+#nullable enable
+
+using OSStatus = System.Int32;
+using SecIdentityRef = System.IntPtr;
+using CFArrayRef = System.IntPtr;
+using NSDataRef = System.IntPtr;
+using NSStringRef = System.IntPtr;
+
+namespace CoreWlan {
+#if NET
+	[SupportedOSPlatform ("maccatalyst15.0")]
+#else
+	[MacCatalyst (15,0)]
+#endif
+	public static partial class CWKeychain {
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainCopyWiFiEAPIdentity (CWKeychainDomain domain, NSDataRef ssid, out SecIdentityRef identity);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool FindWiFiEAPIdentity (CWKeychainDomain domain, NSData ssid, out SecIdentity? identity)
+		{
+			identity = null;
+			IntPtr outPtr = IntPtr.Zero;
+			var result = CWKeychainCopyWiFiEAPIdentity (domain, ssid.GetHandle (), out outPtr);
+			if (result == 0)
+			{
+				identity = new SecIdentity (outPtr, true);
+			}
+
+			return result == 0;
+		}
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainDeleteWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSDataRef ssid);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryDeleteWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid)
```

Comment:
In the past, we've run into cases where it's been annoying to diagnose failures when we don't return the error code, so I think we should do something like this:

```suggestion
		public static bool TryDeleteWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid, out OSError error)
```

possibly as a separate overload (this goes for all the Try* methods here).

Reply:
I have added an overload to each of the methods that will add access to the OSError + an overload with string if possible.